addons/project/models/project.py

```diff
@@ -911,11 +919,82 @@ def _show_profitability(self):
     def _get_profitability_aal_domain(self):
         return [('account_id', 'in', self.analytic_account_id.ids)]
 
-    def _get_profitability_items(self, with_action=True):
+    def action_aal(self, domain=None, res_id=False):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id("analytic.account_analytic_line_action_entries")
+        action['domain'] = domain or []
+        action['context'] = {
+            'grid_anchor': (datetime.now()).strftime('%Y-%m-%d'),
+            'grid_range': 'year',
+            'group_by_date': True,
+        }
+        if res_id:
+            action['view_mode'] = 'form'
+            action['res_id'] = res_id
+        return action
+
+    def _get_action_for_profitablity_section(self, record_ids, name):
+        self.ensure_one()
+        args = [[('id', 'in', record_ids)]]
+        if len(record_ids) == 1:
+            args.append(record_ids[0])
+        return {'name': name, 'type': 'object', 'args': json.dumps(args)}
+
+    def _get_domain_aal_with_no_move_line(self):
+        """ this method is used in order to overwrite the domain in sale_timesheet module. Since the field 'project_id' is added to the "analytic line" model
+        in the hr_timesheet module, we can't add the condition ('project_id', '=', False) here. """
+        return [('account_id', '=', self.analytic_account_id.id), ('move_line_id', '=', False)]
+
+    def _get_items_from_aal(self, with_action=True):
+        domain = self._get_domain_aal_with_no_move_line()
+        aal_other_search = self.env['account.analytic.line'].sudo().search_read(domain, ['id', 'amount'])
+        if not aal_other_search:
+            return None
+        total_revenues = total_costs = 0.0
+        cost_ids = []
+        revenue_ids = []
+        for aal in aal_other_search:
+            aal_amount = aal['amount']
+            if aal_amount < 0.0:
+                total_costs += aal_amount
+                cost_ids.append(aal['id'])
+            else:
+                total_revenues += aal_amount
+                revenue_ids.append(aal['id'])
+        costs_action = None
+        revenues_action = None
+        # The account.groupe_account_readonly is set in the account_accountant module. There's no bridge between this module and the sale_project.
+        # Therefore, we add this security check-up here to ensure that the user has access to accounting.
+        if with_action and self.user_has_groups('account.group_account_readonly'):
```

Comment:
The commentary is very misleading, i should update it. The idea is that without the account_accountant module this code should never be reached. Without accounting, it is not possible to create orphans AAL for that section, and thus the function will return None before reaching the if user.has_group(). Though, i did not think about the case of a user removing the module, in this case the error could indeed appear since AAL would be present but not the group. To prevent the error we could put the code in the new project_account module. That way we have a guarantee to have the 'group_account_readonly' (since the group is defined in account, then over written in account_accountant).  The reason the code was in project before is because sale_project was not necessary to have the use case. Accounting + Project were enough, but there was no bridge between the two.

Reply:
Okay but why not override the method in project_account to change the with_action parameter according to `self.user_has_groups('account.group_account_readonly')`. 🤔

Reply:
Yeah, it is possible to do that too. But i think its better to put the method in the "project_account" module because the use case can not be executed without the "account" module installed, so i dont see the point of putting the code in the "project" module, since it is gonna be useless until the user install "account" module.